index.rst

```diff
@@ -58,24 +58,216 @@ We describe the design and implementation of the LSST Alert Distribution System,
+Confluent platform release 4.1.1.
+The producer used for generating and sending data to Kafka and
+template scripts for consumers of the stream are provided in the GitHub
+repository at https://github.com/lsst-dm/alert_stream,
+which can also be built as a Docker image and deployed as containers.
 
 Alert Filtering
 ================
 
+Alert filters can be written using simple Python functions that act
+solely on the contents of the alert packets.
+Each filter can run separately and isolated in its own container.
+A filter is constructed as a consumer of the Kafka topic containing the
+full stream and a producer back to a Kafka topic for the filtered stream.
+This filtered stream can then be read by another consumer or directed
+to another output method for storage.
+
+Currently, filters are defined in ``lsst.alert.stream.filters`` in the
+alert_stream repository.
+Each filter is constructed as a class with a filter function.
+Below is an example of code defining a simple filter.
+
+.. code-block:: Python
+
+  class Filter001(AlertFilter):
+      def filter(self, alert):
+          if ((alert['diaSource']['snr'] > 5) &
+              (alert['diaSource']['diffFlux'] > 0.00003631)):  # 20th mag
+              return True
+          else:
+              return False
+
+Filters inherit from an AlertFilter base class.
+When the filter class is called on each alert, the visit ID is read.
+Up to 20 alerts per visit that pass the filter, i.e., return True, are
+forwarded to a topic for the filtered alerts, named for the class.
+In the above example, alerts are sent to a new topic named "Filter001."
+The filter detects the beginning of a new visit when the visit ID
+changes from one alert to the next.
+This assumes that all alerts from a visit will be received before
+the next visit's alerts arrive.
+Otherwise, more than 20 alerts may pass through the filter.
+
 
 Alert Database
 ==============
 
 Deployment
 ===========
 
-.. Add content here.
-.. Do not include the document title (it's automatically added from metadata.yaml).
+Deployment of the alert distribution system including filtering with
+the mini-broker is shown in the figure below.
+Content of the alerts is created in science pipelines, potentially
+parallelized by CCD.
+In the alert_stream repo, AlertProducers serialize Python dicts
+into Avro format when alerts are sent to a central Kafka broker,
+or cluster of Kafka brokers acting as one unit.
+For testing, a single AlertProducer can be deployed from the
+alert_stream repo that sends local Avro alert visit files to Kafka.
+Each Kafka broker in this central hub is deployed on its own
+node either with Zookeeper on the same node or Zookeeper on a separate
+node with no other services running.
+The main Kafka system streams to and feeds downstream community broker
+consumers and sinks to the alert database.
+
+
+.. figure:: deployment-diagram.png
+
+
+The central Kafka system feeds the mini-broker filtering system,
```

Comment:
I think you should describe in more detail why you chose this architecture & some of the bandwidth issues we're trying to solve.

Reply:
In particular it would be worth describing why we moved away from the architecture described in DMTN-081.

Reply:
Added text here at the beginning of the deployment section.